src/mono/dbi/cordb-eval.cpp

```diff
@@ -0,0 +1,259 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+// File: CORDB-EVAL.CPP
+//
+
+#include <cordb-appdomain.h>
+#include <cordb-eval.h>
+#include <cordb-function.h>
+#include <cordb-process.h>
+#include <cordb-thread.h>
+#include <cordb-value.h>
+#include <cordb.h>
+
+#include "corerror.h"
+#include "metamodel.h"
+#include "metamodelpub.h"
+#include "rwutil.h"
+#include "stdafx.h"
+
+CordbEval::CordbEval(Connection* conn, CordbThread* thread) : CordbBaseMono(conn)
+{
+    this->m_pThread = thread;
+    if (thread)
+        thread->InternalAddRef();
+    m_pValue    = NULL;
+    m_commandId = -1;
+}
+
+CordbEval::~CordbEval()
+{
+    if (m_pThread)
+        m_pThread->InternalRelease();
+}
+HRESULT STDMETHODCALLTYPE CordbEval::CallParameterizedFunction(ICorDebugFunction* pFunction,
+                                                               ULONG32            nTypeArgs,
+                                                               ICorDebugType*     ppTypeArgs[],
+                                                               ULONG32            nArgs,
+                                                               ICorDebugValue*    ppArgs[])
+{
+    conn->GetProcess()->Stop(false);
```

Comment:
This does not sound right .... is there a reason why the runtime is not already stopped when `CallParameterizedFunction` is called?

Reply:
The runtime is already stopped but on mono side we have a "automatic" continue after eval, but vsdbg also sends a continue. So to avoid continue before receive vsdbg continue, I forced another stop. (On mono side we have a counter and only when it get on 0 it's really resumed)

Reply:
> "automatic" continue after eval

'after eval' means part of the runtime handler that processes the 'INVOKE_METHOD' command below?

Reply:
Yes.